### DISCUSSION TITLE
Add `collection` parameter to `com.atproto.sync.getRepo`

### DISCUSSION THREAD

**Discussion by DavidBuchanan314** [bluesky-social/atproto#3063] [Protocol (atproto)]
It should be possible for a PDS (or relay) to efficiently serve a slice of the repo CAR that contains

a) all the records in a collection
b) all the MST blocks needed to verify them

This would be especially useful for anyone trying to backfill the whole network, but who only cares about a particular collection (say, follows)

It should also be possible to combine the `since` and `collection` parameters (although I haven't thought about how I'd implement it yet, maybe there's some hidden complexity there I haven't considered)

**Comment by DavidBuchanan314** [comment]
Bonus feature - allow a single wildcard at the end of the collection parameter, e.g. `app.bsky.*` (this is efficient to answer since the matching collections are guaranteed to be adjacent in the MST)

**Comment by DavidBuchanan314** [comment]
This feature appears to be included in the Sync 1.1 proposal https://github.com/bluesky-social/proposals/pull/70/files#diff-38ffde66e809b79e3068b7ca59a7a4c8127db5dd3c709a8bb7e654eeb221f7b8R157

**Reply by bnewbold** [reply]
The proposal side-steps the question of `since`, for now. We might deprecate the `since` functionality, to keep PDS implementations simpler? It might be useful to minimize the amount of data transferred; on the other hand keeping track and verifying makes code more complex on all sides.

We would want prefix or wildcard behavior, as you mention above.

Also worth noting somewhere: in some cases, with a fixed record key, this boils down to just the "get record proof" endpoint.

**Reply by mary-ext** [reply]
`since` seems worthwhile to keep around, it enables incremental sync without going through Firehose for an offline-first app, I think not being able to do `since` and `collection` filter might be okay though

**Reply by bnewbold** [reply]
that style of incremental sync only works for consuming services which have a full copy of the repo MST (like a complete CAR file export, in full). with sync v1.1, we mostly expect folks to *not* retain full MST copies, or do the "diff" style manipulation. instead, we expect folks to re-download the full CAR file, but have it be in-order for easy streaming processing (so it doesn't need to be loaded in memory).

requiring PDS implementations to support `since` is a big ask: they need to retain additional metadata for every single node and record, and have relatively efficient indexing to fetch it. and the mechanism ends up not working after account migrations anyways. If it everybody has to deal with the situation that the PDS might just return the full CAR file for `since` requests, we thought we might as well just optimize things for that code path, and simplify PDS implementations while we are at it. I'm definitely feeling like the simplest compliant PDS implementation ends up being quite complicated, and want to reduce what is actually "required" versus "nice to have". 

It does feel like the `since` behavior could be useful for archival and periodic/incremental backup use-cases. But I kind of think this should end up being an (optional) separate endpoint which PDS implementations can decide to implement or not. Having a parameter on such a core/important endpoint be optional feels confusing to me.
